src/plugins/listops/append_operation.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018 Weile Wei
+// Copyright (c) 2018 R. Tohid
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/ir/ranges.hpp>
+#include <phylanx/plugins/listops/append_operation.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    match_pattern_type const append_operation::match_data = {
+        hpx::util::make_tuple("append",
+            std::vector<std::string>{"append(_1, _2)"},
+            &create_append_operation, &create_primitive<append_operation>)};
+
+    ///////////////////////////////////////////////////////////////////////////
+    void append_operation::append_element(
+        std::vector<primitive_argument_type> & result,
+        primitive_argument_type && rhs) const
+    {
+        if (is_list_operand_strict(rhs))
```

Comment:
What are the intended semantics here? Which of the following is what you'd like to achieve:
```
    append(list(1, 2, 3), list(4, 5, 6)) == list(1, 2, 3, 4, 5, 6)
```
or
```
    append(list(1, 2, 3), list(4, 5, 6)) == list(1, 2, 3, list(4, 5, 6))
```
?

Reply:
FWIW, Python exposes the latter semantics:
```
>>> l = [1, 2, 3]       
>>> l.append([3, 4, 5]) 
>>> l                   
[1, 2, 3, [3, 4, 5]]    
```